src/xarray_regrid/utils.py

```diff
@@ -357,6 +380,7 @@ def format_lon(
         if right_pad:
             lon_vals[-right_pad:] = source_lon.values[:right_pad] + 360
         obj = update_coord(obj, lon_coord, lon_vals)
+        obj = ensure_monotonic(obj, lon_coord)
```

Comment:
This is mostly unrelated but I noticed I was getting an error in the `.linear` method under certain grids because this padding can also create duplicate coordinates. As a future todo it would be good to try and minimize even further the number of reindexing steps since these can create copies.